proposals/NNNN-noncopyable-generics.md

```diff
@@ -0,0 +1,975 @@
+`consuming`, or `inout`, when it appears as the type of a function's parameter.
+For details on these parameter ownership modifiers, see
+[SE-377](0377-parameter-ownership-modifiers.md).
+
+### Structs and enums
+Struct and enum types default to implicitly conforming to `Copyable`:
+
+```swift
+struct DataSet /* : Copyable */ {
+  var samples: [Double]
+}
+```
+
+As with any other generic parameter, there is a `Copyable` constraint on the
+`Elm` in this `List` too:
+
+```swift
+enum List<Elm> /* : Copyable where Elm: Copyable */ {
+  case empty
+  indirect case elm(Elm, List<Elm>)
+}
+```
+
+As in SE-390, a generic struct or enum can use the inverse `~Copyable` to
+remove the implicit `Copyable` conformance and constraint:
+
+```swift
+enum List<Elm: ~Copyable>: ~Copyable { /* ... */ }
+```
+
+Once `Elm` does not have a `Copyable` constraint, a noncopyable type like
+`FileDescriptor` can be substituted in `List`, in addition to copyable ones.
+
+#### Automatic synthesis of conditional `Copyable` conformance
+A common reason for having a type parameter is that values of that generic type
+will be stored somewhere within the parameterized struct or enum type. Because
+of `Copyable`'s containment rule, that implies the parameterized type itself
+is commonly noncopyable if any of its type parameters is noncopyable. This
+observation is one of the guiding principles behind the rules in this section.
+
+Using the inverse `~Copyable` on any type parameter of a struct or enum, without
+specifying the copying behavior for the parameterized type, will automatically
+synthesize a conditional conformance to `Copyable`:
+
+```swift
+// A conditionally-copyable pair.
+struct Pair<Elm: ~Copyable> {
+  // signature <Elm>
+  func exchangeFirst(_ new: consuming Elm) -> Elm? { /* ... */ }
+}
+/* extension Pair: Copyable where Elm: Copyable {} */
+
+// Pair<FileDescriptor> is NOT Copyable
+// Pair<Int> is Copyable
+
+// A conditionally-copyable factory.
+struct Factory<Product: ~Copyable, Token: ~Copyable, Policy> {
+  var seed = 0
+
+  // signature <Product, Token, Policy where Policy : Copyable>
+  mutating func produce(_ payment: consuming Token) -> Product { /* ... */ }
+}
+/* extension Factory: Copyable where Product: Copyable, Token: Copyable {} */
+```
+
+Notice that despite this `Factory` not storing any noncopyable data, it does not
+have an implicit unconditional `Copyable` conformance because one of its type
+parameters has removed its default `Copyable` constraint. Instead, a
+conditional conformance was synthesized to make `Factory` copyable once those
+type parameters are `Copyable`.
+
+The synthesis procedure generates the `where` clause of the conditional
+`Copyable` conformance by enumerating _all_ of the type parameters and requiring
+each one to be `Copyable`. It does not matter whether the parameter is used in
+the storage of the `Factory` struct.
+
+One case where this synthesis procedure produces an invalid conformance is
```

Comment:
this feels like it is just one more case where you need control over the synthesis, like the subsequent paragraphs, but is instead special casted just because it's not _optional_ to use this control, it's mandatory.